aiosmtpd/smtp.py

```diff
@@ -469,18 +681,19 @@ async def smtp_HELP(self, arg):
                 return
             code = 501
         commands = []
-        for name in dir(self):
-            if not name.startswith('smtp_'):
-                continue
-            method = getattr(self, name)
+        for name, method in self._smtp_methods.items():
             if self._syntax_available(method):
-                commands.append(name.lstrip('smtp_'))
+                commands.append(name)
```

Comment:
As you can see, caching all those `getattr`s results in a simpler code. Should also be much faster because we can drop the two `str` methods there.